crates/proof-of-sql/src/sql/proof_plans/dyn_proof_plan.rs

```diff
@@ -43,7 +38,6 @@ impl<C: Commitment> ProofPlan<C> for DynProofPlan<C> {
     ) -> Result<(), crate::base::proof::ProofError> {
         match self {
             DynProofPlan::Projection(expr) => expr.count(builder, accessor),
-            DynProofPlan::Filter(expr) => expr.count(builder, accessor),
             DynProofPlan::GroupBy(expr) => expr.count(builder, accessor),
             DynProofPlan::DenseFilter(expr) => expr.count(builder, accessor),
```

Comment:
Can we rename `DenseFilter` to `Filter` in this PR as well? Keep this PR as one of the commits, and then do the rename in another commit (or two).
That way, there is only one breaking change.